### PR TITLE
Prepare for 11.0.0 release

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/3012

 # Rationale for this change

# What changes are included in this PR?

- The first commit has docs update
- The second commit bumps the crate versions

# Are there any user-facing changes?

Yes, version number changed.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -22,14 +22,6 @@ Apache Arrow DataFusion
 Table of Contents
 =================
 
-.. _toc.usage:
-
-.. toctree::
-   :maxdepth: 1
-   :caption: Supported Environments
-
-   Rust <https://docs.rs/crate/datafusion/>
```
Comment (andygrove) [inline docs/source/index.rst:31]: This is the very first link on the page and takes the user to a completely different web site. I moved it down into the user guide.

File: dev/release/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 ### Major Release
 
-DataFusion typically has major releases from the `master` branch every 3 months, including breaking API changes.
+DataFusion typically has major releases from the `master` branch every 4 weeks, including breaking API changes.
```
Comment (alamb) [inline dev/release/README.md:26]: 🚄

File: docs/source/user-guide/sql/ddl.md [hunk 3]
```diff
@@ -30,9 +30,17 @@ STORED AS PARQUET
...
-provide schema information for CSV files since DataFusion does not automatically infer the schema when using SQL
-to query CSV files.
+CSV data sources can also be registered by executing a `CREATE EXTERNAL TABLE` SQL statement. The schema will be
+inferred based on scanning a subset of the file.
+
+```sql
+CREATE EXTERNAL TABLE test
+STORED AS CSV
+WITH HEADER ROW
+LOCATION '/path/to/aggregate_test_100.csv';
```
Comment (alamb) [inline docs/source/user-guide/sql/ddl.md:40]: I didn't realize you could do this 🤔 

I actually tried this particular example and it doesn't work: 

```
❯ CREATE EXTERNAL TABLE test
STORED AS CSV
WITH HEADER ROW
LOCATION '../testing/data/csv/aggregate_test_100.csv';
0 rows in set. Query took 0.102 seconds.
❯ select * from test;
ArrowError(ParseError("Error while parsing value 14857091259186476033 for column 9 at line 3"))

```

Comment (alamb) [reply docs/source/user-guide/sql/ddl.md:40]:
```suggestion
LOCATION '/path/to/aggregate_simple.csv';
```

But an alternate worked great: 

```sql
❯ CREATE EXTERNAL TABLE test
STORED AS CSV
WITH HEADER ROW
LOCATION '../datafusion/core/tests/aggregate_simple.csv';
0 rows in set. Query took 0.014 seconds.
❯ select * from test;
+---------+----------------+-------+
| c1      | c2             | c3    |
+---------+----------------+-------+
| 0.00001 | 0.000000000001 | true  |
| 0.00002 | 0.000000000002 | false |
| 0.00002 | 0.000000000002 | false |
| 0.00003 | 0.000000000003 | true  |
| 0.00003 | 0.000000000003 | true  |
| 0.00003 | 0.000000000003 | true  |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
+---------+----------------+-------+
```

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks for all the work to prepare the release @andygrove 👍

File: docs/source/user-guide/sql/ddl.md (hunk 3 above)
Comment (andygrove) [reply docs/source/user-guide/sql/ddl.md:40]: Thanks for testing that. Filed https://github.com/apache/arrow-datafusion/issues/3174

**Comment by andygrove** [comment]
@alamb Now with CHANGELOG. I will go ahead and merge if there are no objections (once CI passes)

File: datafusion/CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,250 @@
 
 # Changelog
 
+## [11.0.0](https://github.com/apache/arrow-datafusion/tree/11.0.0) (2022-08-16)
+
+[Full Changelog](https://github.com/apache/arrow-datafusion/compare/10.0.0-rc1...11.0.0)
+
+**Breaking changes:**
```
Comment (alamb) [inline datafusion/CHANGELOG.md:26]: this list seems a little light 😆  but I don't think we should spend too much time cleaning this up

**Review by alamb — APPROVED** [review]
👍

**Pull request merged**